operator-common/src/main/java/io/strimzi/operator/common/model/PemAuthIdentity.java

```diff
@@ -0,0 +1,113 @@
+ */
+public class PemAuthIdentity {
+
+    private final byte[] pemPrivateKey;
+    private final byte[] pemCertificateChainBytes;
+    private final X509Certificate pemCertificateChain;
+    private String secretName;
+    private String secretNamespace;
+
+    /**
+     * @param secret Kubernetes Secret containing the Cluster Operator public and private key
+     * @param secretKey Key in the Kubernetes Secret that is associated with the requested identity
+     */
+    private PemAuthIdentity(Secret secret, String secretKey) {
+        Optional.ofNullable(secret)
+                .map(Secret::getMetadata)
+                .ifPresent(objectMeta -> {
+                    secretName = objectMeta.getName();
+                    secretNamespace = objectMeta.getNamespace();
+                });
+        pemPrivateKey = Util.decodeFromSecret(secret, String.format("%s.key", secretKey));
+        pemCertificateChainBytes = Util.decodeFromSecret(secret, String.format("%s.crt", secretKey));
+        pemCertificateChain = validateCertificateChain(secretKey);
+    }
+
+    /**
+     * Create a new instance of PemAuthIdentity that represents the identity of the
+     * cluster operator during TLS client authentication. This also validates the provided
+     * Secret contains a valid certificate chain.
+     *
+     * @param secret Kubernetes Secret containing the client authentication identity
+     *
+     * @return PemAuthIdentity to use as the client authentication identity during TLS authentication
+     */
+    public static PemAuthIdentity clusterOperator(Secret secret) {
+        return new PemAuthIdentity(secret, "cluster-operator");
+    }
+
+    /**
+     * Create a new instance of PemAuthIdentity that represents the identity of the
+     * entity (i.e. user or topic) operator during TLS client authentication. This also validates
+     * the provided Secret contains a valid certificate chain.
+     *
+     * @param secret Kubernetes Secret containing the client authentication identity
+     *
+     * @return PemAuthIdentity to use as the client authentication identity during TLS authentication
+     */
+    public static PemAuthIdentity entityOperator(Secret secret) {
+        return new PemAuthIdentity(secret, "entity-operator");
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a X509Certificate
+     */
+    public X509Certificate pemCertificateChain() {
+        return pemCertificateChain;
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a byte array
+     */
+    public byte[] pemCertificateChainBytes() {
+        return pemCertificateChainBytes;
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a String
+     */
+    public String pemCertificateChainString() {
+        return Util.decodeToString(pemCertificateChainBytes);
```

Comment:
I find the `decodeToString` name deeply confusing, for similar reasons as I mentioned above ("String" isn't enough to tell me what the content is intended to be). `encodeToBase64` would make sense, if that's what it's doing.